### PR TITLE
inits community resources

### LINKED ISSUE #78: [External Resource] Basics
## Goals

- Should be included in the Pagefind search
- Should have categories similar to Astro Docs
- Should be possible replacement for Astro Docs https://docs.astro.build/en/community-resources/content/ & https://docs.astro.build/en/community-resources/talks/
- Should be possible for community to contribute
- Should be easy to find on the sidebar

### PR DESCRIPTION
**Pull request by alexanderniebuhr**
closes #78

In this pull request, I've made strategic decisions to optimize our external resource presentation. Below are the key decisions and their context:

1. **Adopting a Content Collection Approach:**
   - **Rational:** I've decided against a single large `.mdx` file in favor of a content collection model. This approach facilitates future development of API endpoints, enabling external referencing of specific resources only. Although this introduces some additional steps for users to add their resource, but the use of a simple `.json` file mitigates that.

2. **Utilization of `LinkCards`:**
   - **Current Implementation:** For the time being, I am leveraging `<LinkCard />` due to their existing aesthetic and functional design.
   - **Future Plans:** I aim to develop custom cards that include additional metadata such as author names, publication times, likes, and views. This enhancement will support a richer user experience and better content engagement.

3. **Content Synchronization with Astro Documentation:**
   - **Interim Strategy:** I have temporarily copied categories and external content from the Astro documentation. As we discuss potential replacements, it is crucial to ensure alignment with Astro to maintain consistency and integration fidelity.

### CODE REVIEW

File: src/components/ResourceCards.astro [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    | "markdown"
+    | "performance"
+    | "utilities"
+    | "animation"
+    | "i18n"
+    | "db";
+}
+
+const { category } = Astro.props;
+const allResources = await getCollection("resources", ({ data }: any) => {
```
Comment (alexanderniebuhr) [inline src/components/ResourceCards.astro:22]: Why does the type infer doesn't work?

Comment (alexanderniebuhr) [reply src/components/ResourceCards.astro:22]: The `examples` messed around with ts.

File: astro.config.ts [hunk 2]
```diff
@@ -31,11 +36,23 @@ export default defineConfig({
...
       },
     }),
   ],
+  vite: {
+    resolve: {
+      alias: {
+        '~': resolve(dirname(fileURLToPath(import.meta.url)), './src'),
+      },
+    },
+  }
```
Comment (florian-lefebvre) [inline astro.config.ts:57]: Why is this needed? Aren't we using something in the tsconfig?

Comment (alexanderniebuhr) [reply astro.config.ts:57]: This is a change while I tried to fix the `astro check` issues. This would allow us to get rid of `"baseUrl": "."`, not sure if you care about that?

Comment (florian-lefebvre) [reply astro.config.ts:57]: Not at all! Just curious

**Review by florian-lefebvre — APPROVED** [review]

File: src/content/docs/resources/educational.mdx [hunk 3]
```diff
@@ -0,0 +1,59 @@
+---
+title: Community Ressources
```
Comment (BryceRussell) [inline src/content/docs/resources/educational.mdx:2]:
```suggestion
title: Community Resources
```

File: src/content/docs/resources/educational.mdx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+import ResourceCards from '~/components/ResourceCards.astro'
+
+See guided examples of adding features to your Astro project, submitted by our community members!
+
+[Astro recipes](https://docs.astro.build/en/recipes/) are typically short, focused how-to guides that walk a reader through completing a working example of a specific task. Recipes are a great way to add new features or behavior to your Astro project by following step-by-step instructions!
+
+Other guides might explain concepts related to an area of content, such as using images or working with MDX.
+
+:::tip[Add your own!]
```
Comment (sarah11918) [inline src/content/docs/resources/educational.mdx:13]: Just reminding that this is a placeholder and you'll eventually want your own process here. If you're not having one big `.mdx` page, then people won't simply be editing this page an adding a link.

Have you thought about whether you want people to submit via a form and *you* create the page for them, or whether you will want people to e.g. "create a new `.mdx` file under `src/content/resources/`"?

File: src/content/resources/blog-images-rss.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+  "category": "images",
+  "title": "Add blog post images to your Astro RSS Feed",
+  "link": "https://webreaper.dev/posts/astro-rss-feed-blog-post-images/"
```
Comment (sarah11918) [inline src/content/resources/blog-images-rss.json:4]:
```suggestion
  "link": "https://webreaper.dev/posts/astro-rss-feed-blog-post-images/"
  "author": "Boston343"
```
Wondering about future-proofing this, in the case you want these to look even fancier e.g. with a Link to the author's GitHub profile or displaying their profile picture.

Fun fact! Just like someone's GitHub profile url is `https://github.com/<username>`, their profile picture is available at `https://github.com/<username>.png`!

So, storing just the username gives you opportunities for whatever aspect of their profile you might want to do something with.

If you'd like this, then I can be tracking these down for you.

**Review by sarah11918 — COMMENTED** [review]
This looks amazing!

I would like us to figure out how to best coordinate between the two sites, but for now this is an excellent start. Just a couple of thoughts I had below!

File: src/content/resources/blog-images-rss.json (hunk 5 above)
Comment (alexanderniebuhr) [reply src/content/resources/blog-images-rss.json]: Yeah this sounds great!
We do want author information, we have a working branch for adding that here https://github.com/astrolicious/astro-tips.dev/pull/79
So I think we should use the same solution, but if we can already track them down, that would be amazing

File: .gitpod.yml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# This configuration file was automatically generated by Gitpod.
+# Please adjust to your needs (see https://www.gitpod.io/docs/introduction/learn-gitpod/gitpod-yaml)
+# and commit this file to your remote git repository to share the goodness with others.
+
+# Learn more from ready-to-use templates: https://www.gitpod.io/docs/introduction/getting-started/quickstart
+
+tasks:
+  - init: pnpm install && pnpm run build
+    command: pnpm run start
```
Comment (sarah11918) [inline .gitpod.yml:9]:
```suggestion
tasks:
  - init: pnpm install
    command: pnpm run start
```
Just checking whether you intended this, or even to commit this at all, because I know this is usually the typical Gitpod start up file when you don't have one.

It doesn't matter as much *now* when your site is small, but I have my `gitpod.yml` specifically set NOT to build the docs site when I open it because of how long it takes!!  😅

Comment (alexanderniebuhr) [reply .gitpod.yml:9]: I just added everything from the GitPod, without looking at this file, not sure if we want it or not. Might be helpful if we use GitPod in the future too.. 

cc @florian-lefebvre ??

Comment (florian-lefebvre) [reply .gitpod.yml:9]: i'm not familiar with gitpod at all, if this file helps us start more gitpod sessions in the future then let's keep it

Comment (alexanderniebuhr) [reply .gitpod.yml:9]: @sarah11918 can you enlighten us about GitPod a little bit?

Comment (sarah11918) [reply .gitpod.yml:9]: Yes I can!

When you start a Gitpod, this file will be created for you with their defaults. If you just blanket commit everything, then you commit this file.

So you don't *need* this file at all! Gitpod will always create a new one for you! But, in doing so, it's always Gitpod's default which is to both install and build your project. (Which, for docs, is a BAD IDEA because of how long it takes.)

That's why docs *does* use a file, not to make it work, but rather to override the defaults when you don't have one.

Comment (alexanderniebuhr) [reply .gitpod.yml:9]: Ok then I think it makes sense to keep it :)
